apps/gitness/src/pages-v2/repo/stores/repo-branches-store.ts

```diff
@@ -1,21 +1,100 @@
 import { create } from 'zustand'
 
-import { BranchSelectorListItem, BranchSelectorTab, IBranchSelectorStore } from '@harnessio/ui/views'
+import {
+  CalculateCommitDivergenceOkResponse,
+  TypesBranchExtended,
+  TypesRepository
+} from '@harnessio/code-service-client'
+import { BranchData, BranchSelectorListItem, BranchSelectorTab, CommitDivergenceType } from '@harnessio/ui/views'
 
-export const useRepoBranchesStore = create<IBranchSelectorStore>(set => ({
-  selectedBranchTag: { name: '', sha: '' },
-  setSelectedBranchTag: (selectedBranchTag: BranchSelectorListItem) => set({ selectedBranchTag }),
+import { timeAgoFromISOTime } from '../../../pages/pipeline-edit/utils/time-utils'
+import { PageResponseHeader } from '../../../types'
 
-  selectedBranchType: BranchSelectorTab.BRANCHES,
-  setSelectedBranchType: (selectedBranchType: BranchSelectorTab) => set({ selectedBranchType }),
+export interface IBranchSelectorStore {
+  // states
+  selectedBranchTag: BranchSelectorListItem
+  selectedBranchType: BranchSelectorTab
+  branchList: BranchData[]
+  tagList: BranchSelectorListItem[]
+  spaceId: string
+  repoId: string
+  defaultBranch: string
+  branchDivergence: CommitDivergenceType[]
+  xNextPage: number
+  xPrevPage: number
+  page: number
 
-  branchList: [{ name: '', sha: '' }],
-  setBranchList: (branchList?: BranchSelectorListItem[]) => set({ branchList }),
-
-  tagList: [{ name: '', sha: '' }],
-  setTagList: (tagList: BranchSelectorListItem[]) => set({ tagList }),
+  //actions
+  setSelectedBranchTag: (selectedBranchTag: BranchSelectorListItem) => void
+  setSelectedBranchType: (selectedBranchType: BranchSelectorTab) => void
+  setTagList: (tagList: BranchSelectorListItem[]) => void
+  setSpaceIdAndRepoId: (spaceId: string, repoId: string) => void
+  setBranchList: (
+    branches: TypesBranchExtended[],
+    divergence?: CalculateCommitDivergenceOkResponse,
+    defaultBranch?: string
+  ) => void
+  setDefaultBranch: (metadata: TypesRepository) => void
+  setBranchDivergence: (divergence: CalculateCommitDivergenceOkResponse) => void
+  setPage: (page: number) => void
+  setPaginationFromHeaders: (headers?: Headers) => void
+}
 
+export const useRepoBranchesStore = create<IBranchSelectorStore>(set => ({
+  // initial state
+  defaultBranch: '',
+  branchDivergence: [],
   spaceId: '',
   repoId: '',
-  setSpaceIdAndRepoId: (spaceId: string, repoId: string) => set({ spaceId, repoId })
+  tagList: [{ name: '', sha: '' }],
```

Comment:
tags should be in a separate store maybe

Reply:
Sure...separate PR maybe?